telemetry/src/main/java/com/newrelic/telemetry/TelemetryClient.java

```diff
@@ -164,11 +165,21 @@ private void scheduleBatchSend(
       long waitTime,
       TimeUnit timeUnit,
       Backoff backoff) {
-    executor.schedule(() -> sendWithErrorHandling(sender, batch, backoff), waitTime, timeUnit);
+    try {
+      executor.schedule(() -> sendWithErrorHandling(sender, batch, backoff), waitTime, timeUnit);
+    } catch (RejectedExecutionException e) {
+      LOG.error("Problem scheduling batch.", e);
+    }
   }
 
   private void sendWithErrorHandling(
       BatchSender batchSender, TelemetryBatch<? extends Telemetry> batch, Backoff backoff) {
+    if (executor.isTerminated()) {
```

Comment:
I don't think we want to check the executor here.  If we have reached this line, it means that the executor has invoked us, and we should be ok to send the batch.  If the timing is perfectly wrong, I think this could cause us to lose data.